linkshell_service/linkshell_service.lua

```diff
@@ -0,0 +1,72 @@
+local packets = require('packets')
+local shared = require('shared')
+
+local linkshell = shared.new('linkshell')
+
+-- Does this file need this?
+linkshell.env = {
+    next = next,
+}
+  
+local linkshell.data = {
+    [1] = { message = { }, color = { }, },
+    [2] = { message = { }, },
+}
+
+local handle_color = function(p)
+local data = linkshell.data[1]
+    data.color.red = p.linkshell_red
+    data.color.green = p.linkshell_green
+    data.color.blue = p.linkshell_blue
+end
+
+packets.incoming[0x0CC]:register(function(p)
+    local ls_number = bit.band(p.flags, 0x40) == 0x40 and 2 or 1
+    local data = linkshell.data[ls_number]
+    data.name = p.linkshell_name
+    data.message.timestamp = p.timestamp
+    data.message.author = p.player_name
+    data.message.text = p.message
+    data.message.permissions = p.permissions
+end)
+
+packets.incoming[0x0E0]:register(function(p)
+    local data = linkshell[p.linkshell_number]
+    data.bag_index = p.bag_index
+end)
+
+packets.incoming[0x00D]:register(handle_color)
+packets.incoming[0x037]:register(handle_color)
+packets.incoming[0x0C9]:register(handle_color)
+
+handler_color(packets.incoming[0x00D].last)
```

Comment:
Check if `.last` is `nil` before calling the handler. See what the `account_service` does here.